### PR TITLE
Blazor hosting models UE pass

### PR DESCRIPTION
**Pull request by guardrex**
Addresses #20142
Addresses #19286

[Internal Review Topic](https://review.docs.microsoft.com/en-us/aspnet/core/blazor/hosting-models?view=aspnetcore-3.1&branch=pr-en-us-20868)

### CODE REVIEW

File: aspnetcore/blazor/hosting-models.md
Comment (mkArtakMSFT) [inline aspnetcore/blazor/hosting-models.md:26]: Not sure whether this is better or not. I'll leave it up to you, @guardrex  to judge.

```suggestion
To create a Blazor WebAssembly app in [Visual Studio](https://visualstudio.microsoft.com), use the **Blazor App** > **Blazor WebAssembly App** project template. After selecting the template, you have the option of enabling the `hosted` model by selecting the **ASP.NET Core hosted** check box. This will also generate a server project in the created solution and add sample API interaction call in the client app.
```

File: aspnetcore/blazor/hosting-models.md
Comment (mkArtakMSFT) [inline aspnetcore/blazor/hosting-models.md:42]: This sounds too negative, and comes along as `don't use this model, as it's bad`.

```suggestion
Blazor WebAssembly hosting model has some limitations, which you have to be aware of:
```

File: aspnetcore/blazor/hosting-models.md
Comment (mkArtakMSFT) [inline aspnetcore/blazor/hosting-models.md:79]:
```suggestion
Blazor Server hosting model also has some limitations:
```

File: aspnetcore/blazor/hosting-models.md
Comment (mkArtakMSFT) [inline aspnetcore/blazor/hosting-models.md:90]:
```suggestion
One way to understand Blazor Server apps is to understand how it differs from traditional models for rendering UI in ASP.NET Core apps using Razor views or Razor Pages. Both models use the [Razor language](xref:mvc/views/razor) to describe the HTML content to be rendered, but they significantly differ in how markup is rendered.
```

File: aspnetcore/blazor/hosting-models.md
Comment (mkArtakMSFT) [inline aspnetcore/blazor/hosting-models.md:138]: I think it's worth clarifying in `which` memory the state should remain for the connection to resume without issues.

**Review by mkArtakMSFT — APPROVED** [review]
Thanks @guardrex !

File: aspnetcore/blazor/hosting-models.md
Comment (guardrex) [reply aspnetcore/blazor/hosting-models.md:26]: I'll edit the suggestion to remove "model" so that it doesn't sound like there's a third hosting model for Blazor. Thus far, we've always only defined "hosting model" as *how components are hosted within an app* (WASM or Server). We haven't split WASM into two further "hosting models" (standalone and hosted).

Now that I reflect on this, I think the document should call that concept out. If we define it explicitly, we should 🤞 be able to avoid some of the overloading of the word "hosted" these days. I'll add that on a future commit and leave a comment with the text that I used.

```suggestion
To create a Blazor WebAssembly app in [Visual Studio](https://visualstudio.microsoft.com), use the **Blazor App** > **Blazor WebAssembly App** project template. After selecting the template, you have the option of selecting the **ASP.NET Core hosted** check box to serve the Blazor WebAssembly app to clients from an ASP.NET Core app. The project template for a hosted Blazor WebAssembly solution includes separate server and client projects. The solution includes a sample web API interaction between the server and client projects.
```

File: aspnetcore/blazor/hosting-models.md
Comment (guardrex) [inline aspnetcore/blazor/hosting-models.md:138]:
```suggestion
Blazor Server apps require an active SignalR connection to the server. If the connection is lost, the app attempts to reconnect to the server. As long as the client's state remains in the server's memory, the client session resumes without losing state.
```

Comment (guardrex) [reply aspnetcore/blazor/hosting-models.md]: Done on https://github.com/dotnet/AspNetCore.Docs/pull/20868/commits/2052dc0e45ac31e4866ac02adef2c79a17d68033.

File: aspnetcore/blazor/hosting-models.md
Comment (guardrex) [reply aspnetcore/blazor/hosting-models.md]: Thinking a bit further 🤔 in more detail ...

Thus far based on the original text, "how components are hosted by the app" boils down to how the component C# code is executed ... WASM runtime in the browser versus on the server in an ASP.NET Core app.

Comment (guardrex) [reply aspnetcore/blazor/hosting-models.md]: 🤔 ... I like what we have. The rubber :duck: says that we already scope the word "hosting (model)" fairly well in the opening paragraphs. The bits that I add for the difference between standalone and "hosted" WASM clarify that in that context we're talking about deployment scenarios, and I use "deployment" there.

This was probably all considered (and rejected) in ancient Blazor design times, but an "ASP.NET Core hosted" WASM app might have been called an "ASP.NET Core deployed/deployment" WASM app, which may have eliminated the double context of "host/hosted." Anywho ... that's all water under the bridge at this point. :smile:

Comment (mkArtakMSFT) [reply aspnetcore/blazor/hosting-models.md]: I agree. Please go with your original text.

**Comment by guardrex** [comment]
I'll review once more on Monday morning before merging.

@danroth27 @mkArtakMSFT @javiercn ... Please let me know on my **_'Pings anyone?'_** email (from Friday morning) to what degree you want to look at the updates as I go.

**Comment by guardrex** [comment]
@mkArtakMSFT ... *Ugh!* This became another one of my TL;DR 📚 ...🙈😄, but I need to rubber :duck: through some stuff and inquire on a guidance layout problem **_across topics_**.

One of the things that we well know now is that devs :heart: *click to copy* :heart: and then paste into a CLI shell or into their app from the topics. Since a reader sees how to create a Blazor app here, I place the commands for creating apps into our `dotnetcli` styling on the last commit. **_However_**, I think this coverage here **_is a BIG mistake_** now that I look over all of the topics where we have 'how to create a Blazor app' guidance. Therefore, I plan to remove it from this topic.

Devs reading down the ToC encounter this sequence of guidance for creating new Blazor apps ...

1. [**Build your first Blazor app** tutorial](https://dotnet.microsoft.com/learn/aspnet/blazor-tutorial/create): CLI only.
1. [**Tooling** topic](https://docs.microsoft.com/en-us/aspnet/core/blazor/tooling?view=aspnetcore-5.0&pivots=windows): Windows and macOS tabs show use of VS, while Linux shows use of CLI. This one has an additional delta because it doesn't go by **_tooling_** ... it goes by **_platform_**.
1. **Hosting Models** (on this PR): Up to the latest commit, VS is in one paragraph, and the CLI commands are then shown in the following paragraph.
1. [**ToDo List tutorial**](https://docs.microsoft.com/en-us/aspnet/core/tutorials/build-a-blazor-app?view=aspnetcore-5.0): CLI only.
1. [**SignalR with Blazor WASM** tutorial](https://docs.microsoft.com/en-us/aspnet/core/tutorials/signalr-blazor-webassembly?view=aspnetcore-5.0&tabs=visual-studio): VS for Windows, VS4Mac for macOS, and the CLI for VSCode and CLI tabs.
1. [**Templates** topic](https://docs.microsoft.com/en-us/aspnet/core/blazor/templates?view=aspnetcore-5.0): CLI commands only.

... see what I mean? Just the natural churn of topics over time has led to this 😵 *scatterbrained guidance* 😵.

My proposals ...

1. Tooling topic: Outside of 'Build your first', this is our first explanation of how to create Blazor apps using the various tooling approaches. It's fine that it goes by platform, except I've always been a bit put off by pushing devs away from (**_hiding_**) VSC tooling that they know is xplat. A :heart: *VSC lover* :heart: (on Windows) will find themselves on the Windows pivot only describing VS. One might conclude that you can't even use VSC with Blazor on Windows based on this. Let's have a paragraph below the coverage in each doc pivot that says VSC is xplat and then links to the *Linux* pivot. I'll also add a bit to each pivot on how to create the hosted WASM solution. Let's also call out CLI help options here, which is currently in the *wrong spot* down in the Templates topic (I'll move that bit here).

1. Hosting Models topic: Let's drop 'how to create a new app' guidance from the Hosting Models topic and replace it with a sentence that has a cross-link to the Tooling topic.

1. ToDo List tutorial and SignalR with Blazor WASM tutorial: I recommend that we be consistent. If we leave the ToDo List tutorial as @danroth27 instructed for CLI only (with my proposal for one sentence cross-link to the Tooling topic), shouldn't we handle the SignalR tutorial in the same fashion? Alternatively, we could put the various tooling approaches back into the ToDo List tutorial (making it like the SignalR tutorial) and make the tooling tabs default to the CLI tab. Even if we stick to the CLI for either or both, we've already had one reader write in on the ToDo List tutorial asking, 'Are we back in the Stone Age here? Why can't I use VS for this?' ... and I had to reply, 'Go ahead. It's cool.' I think we need to at least cross-link the Tooling topic if we leave the ToDo List tutorial as CLI only.

1. Templates topic: I think 🤔 that we should refactor this one into *Blazor project structure* (or something like that) and move the little bit at the top on template options to the Tooling topic, which is where I think we can coalesce our 'how to create a Blazor app' coverage (i.e., 'using the templates'). IMO, this topic should be focused on **_what files are in a Blazor app and what do they do_**. I think it works great in that capacity.

I'll make a few updates here to cover the Tooling topic, Hosting Models topic, and Templates topic, but that Item 3 :point_up: requires feedback before I can proceed on that one.

**Comment by guardrex** [comment]
@mkArtakMSFT ... The three-day window is past. Do you want me to make the call on https://github.com/dotnet/AspNetCore.Docs/pull/20868#issuecomment-739982806 and move on? These older PRs are holding me up from another update that I need to make, so I'd like to get these in.

**Pull request merged**
